cmd/cloud_run.go

```diff
@@ -38,12 +83,92 @@ Use the "k6 cloud login" command to authenticate.`,
 			"the k6 cloud run command expects a single argument consisting in either a path to a script or "+
 				"archive file, or the \"-\" symbol indicating the script or archive should be read from stdin",
 		),
-		PreRunE: deprecatedCloudCmd.preRun,
-		RunE:    deprecatedCloudCmd.run,
+		PreRunE: cloudRunCmd.preRun,
+		RunE:    cloudRunCmd.run,
+	}
+
+	thisCmd.Flags().SortFlags = false
+	thisCmd.Flags().AddFlagSet(cloudRunCmd.flagSet())
+	thisCmd.Flags().AddFlagSet(cloudCmd.flagSet())
+
+	return thisCmd
+}
+
+func (c *cmdCloudRun) preRun(cmd *cobra.Command, args []string) error {
+	if c.localExecution {
+		if cmd.Flags().Changed("exit-on-running") {
+			return errext.WithExitCodeIfNone(
+				fmt.Errorf("the --local-execution flag is not compatible with the --exit-on-running flag"),
+				exitcodes.InvalidConfig,
+			)
+		}
+
+		if cmd.Flags().Changed("show-logs") {
+			return errext.WithExitCodeIfNone(
+				fmt.Errorf("the --local-execution flag is not compatible with the --show-logs flag"),
+				exitcodes.InvalidConfig,
+			)
+		}
+
+		return nil
+	}
+
+	if c.linger {
+		return errext.WithExitCodeIfNone(
+			fmt.Errorf("the --linger flag can only be used in conjunction with the --local-execution flag"),
+			exitcodes.InvalidConfig,
+		)
+	}
+
+	return c.deprecatedCloudCmd.preRun(cmd, args)
+}
+
+func (c *cmdCloudRun) run(cmd *cobra.Command, args []string) error {
+	if c.localExecution {
+		return c.runCmd.run(cmd, args)
+	}
+
+	// When running the `k6 cloud run` command explicitly disable the usage report.
+	c.noUsageReport = true
+
+	return c.deprecatedCloudCmd.run(cmd, args)
+}
+
+func (c *cmdCloudRun) flagSet() *pflag.FlagSet {
+	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
+	flags.SortFlags = false
+
+	flags.BoolVar(&c.localExecution, "local-execution", c.localExecution,
+		"executes the test locally instead of in the cloud")
+	flags.BoolVar(
+		&c.linger,
+		"linger",
+		c.linger,
+		"only when using the local-execution mode, keeps the API server alive past the test end",
+	)
+	flags.BoolVar(
+		&c.noUsageReport,
+		"no-usage-report",
+		c.noUsageReport,
+		"only when using the local-execution mode, don't send anonymous stats to the developers",
```

Comment:
It is not in the scope of this PR, but we should adjust texting here since `send anonymous stats to the developers` sounds a bit incomplete and not clear. 

It's probably better to provide the link to documentation, like:
```suggestion
		"only when using the local-execution mode, don't send anonymous usage stats (https://grafana.com/docs/k6/latest/set-up/usage-collection/)",
```

Reply:
I think we just re-used what already existed there https://github.com/grafana/k6/blob/master/cmd/config.go#L33

In theory I'm not against your proposal, but then, we change it in both places, right?

Reply:
Yes, the proposal is relevant for the other place, too, and as I said, it's not the scope of the PR. I'm totally fine with adjusting texting in a separate PR, that's why I've approved this one.